python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -483,6 +526,115 @@ def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationRe
+        for dt in cron_string_iterator(
+            start_timestamp=context.cursor.latest_evaluation_timestamp,
+            cron_string=self.cron_schedule,
+            execution_timezone=self.timezone,
+        ):
+            if dt > context.evaluation_time:
+                break
+            missed_ticks.append(dt)
+        return missed_ticks
+
+    def get_asset_partitions_to_request(
+        self, context: RuleEvaluationContext
+    ) -> AbstractSet[AssetKeyPartitionKey]:
+        missed_ticks = self.missed_cron_ticks(context)
+
+        if not missed_ticks:
+            return set()
+
+        partitions_def = context.asset_graph.get_partitions_def(context.asset_key)
+        if partitions_def is None:
+            return {AssetKeyPartitionKey(context.asset_key)}
+
+        # if all_partitions is set, then just return all partitions if any ticks have been missed
+        if self.all_partitions:
+            return {
+                AssetKeyPartitionKey(context.asset_key, partition_key)
+                for partition_key in partitions_def.get_partition_keys(
+                    current_time=context.evaluation_time
+                )
+            }
+
+        # for partitions_defs without a time component, just return the last partition if any ticks
+        # have been missed
+        time_partitions_def = get_time_partitions_def(partitions_def)
+        if time_partitions_def is None:
+            return {
+                AssetKeyPartitionKey(context.asset_key, partitions_def.get_last_partition_key())
+            }
+
+        missed_time_partition_keys = filter(
+            None,
+            [
+                time_partitions_def.get_last_partition_key(current_time=missed_tick)
+                for missed_tick in missed_ticks
+            ],
+        )
+        # for multi partitions definitions, request to materialize all partitions for each missed
+        # cron schedule tick
+        if isinstance(partitions_def, MultiPartitionsDefinition):
+            return {
+                AssetKeyPartitionKey(context.asset_key, partition_key)
+                for time_partition_key in missed_time_partition_keys
+                for partition_key in partitions_def.get_multipartition_keys_with_dimension_value(
+                    partitions_def.time_window_dimension.name,
+                    time_partition_key,
+                    dynamic_partitions_store=context.instance_queryer,
+                )
+            }
+        else:
+            return {
+                AssetKeyPartitionKey(context.asset_key, time_partition_key)
+                for time_partition_key in missed_time_partition_keys
+            }
+
+    def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationResults:
+        asset_partitions_to_request = self.get_asset_partitions_to_request(context)
+        asset_partitions_by_evaluation_data = defaultdict(set)
+        if asset_partitions_to_request:
+            # TODO: better evaluation data
```

Comment:
in general, I think we try to avoid committing TODOs like this. would be better to track with an issue.